test/browser/browser_asrouter_whatsnewpanel.js

```diff
@@ -0,0 +1,48 @@
+const { PanelTestProvider } = ChromeUtils.import(
+  "resource://activity-stream/lib/PanelTestProvider.jsm"
+);
+const { ToolbarPanelHub } = ChromeUtils.import(
+  "resource://activity-stream/lib/ToolbarPanelHub.jsm"
+);
+
+add_task(async function test_messages_rendering() {
+  const msgs = (await PanelTestProvider.getMessages()).filter(
+    ({ template }) => template === "whatsnew_panel_message"
+  );
+
+  Assert.ok(msgs.length, "FxA test message exists");
+
+  Object.defineProperty(ToolbarPanelHub, "messages", {
+    get: () => Promise.resolve(msgs),
+    configurable: true,
+  });
+
+  await ToolbarPanelHub.enableAppmenuButton();
+
+  const mainView = document.getElementById("appMenu-mainView");
+  UITour.showMenu(window, "appMenu");
+  await BrowserTestUtils.waitForEvent(mainView, "ViewShown");
```

Comment:
Looks like we were clicking too early before the main view was done showing, so internally it was preventing loading of a subpanel.

https://searchfox.org/mozilla-central/rev/c8dba8c886609d0fdb8fae68bab865bf03f1165c/browser/components/customizableui/PanelMultiView.jsm#671-676